### DISCUSSION TITLE
Understanding touch and display pins & signals

### DISCUSSION THREAD

**Discussion by nradina** [Bodmer/TFT_eSPI#2295] [Q&A - General]
Hi everyone!

I'm working on a project using one of the [Ai AudioKit dev boards ](https://www.pschatzmann.ch/home/2021/12/06/the-ai-thinker-audio-kit-experience-or-nothing-is-right/)and trying to include a simple SPI 2.4" touch screen (Ili8341).

When using a standalone ESP32 DEV and screen, all worked great!

While testing the AudioKit board tonight, I noticed the touch portion is working great (using the test_touch_controller example), yet no image (just the usual white screen).

While trying to figure this out, I was curious to understand what pins and signals are used to create the display?
I'm assuming I've made an error with pin assignments or simply wired wrong.

Are there tell tale signs when a specific PIN is not connected correctly?

Just trying to better understand the process.

Thanks for any help!

**Comment by Bodmer** [comment]
You can use this sketch to test the touch controller interface, look at the serial monitor output:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Test%20and%20diagnostics/Test_Touch_Controller

Since the display was working it is best to check all connections again. If you have upgraded the TFT_eSPI library you will need to edit the setup again as it gets overwritten.
